src/Octoshift/GithubClient.cs

```diff
@@ -156,12 +162,24 @@ public virtual async IAsyncEnumerable<JToken> GetAllAsync(string url, Dictionary
             {
                 response.EnsureSuccessStatusCode();
             }
-            else if (response.StatusCode != status)
+            else if (response.StatusCode != status && RateLimitRemaining(headers) != 0)
```

Comment:
Tests still fail if I simulate an HTTP response with a status code of 403 (which could happen with a rate-limited response) — how best to make this not raise an error if we get a 403 and `X-RateLimit-Remaining: 0`?

Reply:
I guess the best would be to also check the error message in case of `403`

Reply:
I think it's because of a misunderstanding of how this if statement is working. In this code `status` is the argument passed in, which could probably be renamed to `expectedStatus`. In the vast majority of circumstances that will have the default value of HttpStatusCode.OK so the else here will never be hit, it will go into the first part of the if statement, EnsureSuccessStatusCode() will run, see a 403 and error out.